pkg/operator/controller/status/controller_test.go

```diff
@@ -664,3 +664,146 @@ func Test_computeOperatorEvaluationConditionsDetectedCondition(t *testing.T) {
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "IngressDoesNotExist",
+				Message: `The "default" ingress controller does not exist.`,
+			},
+		},
+		{
+			description: "default ingresscontroller degraded",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", true),
+					icWithStatus("test", false),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "IngressDegraded",
+				Message: `The "default" ingress controller reports Degraded=True: dummy: dummy.`,
+			},
+		},
+		{
+			description: "default ingresscontroller not degraded",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", false),
+					icWithStatus("test2", true),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionFalse,
+				Reason:  "IngressNotDegraded",
+				Message: `The "default" ingress controller reports Degraded=False.`,
+			},
+		},
+		{
+			description: "default ingresscontroller status unknown",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					ic("default"),
+				},
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionUnknown,
+				Reason:  "IngressDegradedStatusUnknown",
+				Message: `The "default" ingress controller reports Degraded=Unknown.`,
+			},
+		},
+		{
+			description: "default ingresscontroller not degraded but unmanaged gateway api crds exist",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", false),
+				},
+				unmanagedGatewayAPICRDNames: "listenersets.gateway.networking.x-k8s.io",
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "GatewayAPICRDsDegraded",
+				Message: "Unmanaged Gateway API CRDs found: listenersets.gateway.networking.x-k8s.io.",
+			},
+		},
+		{
+			description: "default ingresscontroller degraded and unmanaged gateway api crds exist",
+			state: operatorState{
+				IngressControllers: []operatorv1.IngressController{
+					icWithStatus("default", true),
+				},
+				unmanagedGatewayAPICRDNames: "listenersets.gateway.networking.x-k8s.io",
+			},
+			expectCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorDegraded,
+				Status:  configv1.ConditionTrue,
+				Reason:  "IngressDegradedAndGatewayAPICRDsDegraded",
+				Message: `The "default" ingress controller reports Degraded=True: dummy: dummy.Unmanaged Gateway API CRDs found: listenersets.gateway.networking.x-k8s.io.`,
```

Comment:
Ok, this is the revised format.  The concatenated message is 

> dummy.Unmanaged Gateway API CRDs found: listenersets.gateway.networking.x-k8s.io.

Reply:
And a new degraded message will look something like this if the CRDs also put it in a degraded state:

> "One or more other status conditions indicate a degraded state: DeploymentReplicasMinAvailable=False (DeploymentMinimumReplicasNotMet: 0/2 of replicas are available, max unavailable is 1).Unmanaged Gateway API CRDs found: listenersets.gateway.networking.x-k8s.io."

Reply:
I agree, the readability of the degraded message can be improved.